zng/resolver/context.go

```diff
@@ -341,22 +331,66 @@ func (c *Context) addTypeWithLock(typ zng.Type) {
 	}
 }
 
-// addType adds a new type from a path that could race with another
+// AddType adds a new type from a path that could race with another
 // path creating the same type.  So we take the lock then check if the
 // type already exists and if not add it while locked.
-func (c *Context) addType(t zng.Type) zng.Type {
+func (c *Context) AddType(t zng.Type) zng.Type {
 	c.mu.Lock()
 	defer c.mu.Unlock()
-	key := t.String()
+	key := typeKey(t)
 	id, ok := c.lut[key]
 	if ok {
 		t = c.table[id]
 	} else {
-		c.addTypeWithLock(t)
+		c.addTypeWithLock(key, t)
 	}
 	return t
 }
 
+// LookupTypeRecord returns a zng.TypeRecord within this context that binds with the
+// indicated columns.  Subsequent calls with the same columns will return the
+// same record pointer.  If the type doesn't exist, it's created, stored,
+// and returned.  Columns must contain types this same context.  If you want to
```

Comment:
"Columns must contain types this same context" ?

Reply:
rephrased